### PR TITLE
Bootstrap manifest in client if none was given

### PR DESCRIPTION
**Pull request by mlegore**
Add functionality to call a manifest method on the server in order to
bootstrap the manifest. This pre-supposes that the server has a manifest
method, so those using this functionality should ensure that is the
case, as a convention.

### CODE REVIEW

**Comment by dominictarr** [comment]
hmm, or instead of passing a boolean `bootstrap` what about just pass a `bootstrapCallback` and if that callback is provided, it bootstraps and calls back the setup rpc function. Since ssb-client is already async it wouldn't actually change the api.

also then we don't have to add a reserved word (because users can't have a method named "bootstrap" because it will collide with the built-in bootstrap method.

**Comment by mlegore** [comment]
I've made those changes, I'm open to feedback on how that parameter is passed in, I made sure it works with the current method overloads, but it made passing the parameter a bit awkward.

**Comment by mlegore** [comment]
@dominictarr can you look at this?

File: api.js [hunk 1]
```diff
@@ -29,8 +29,8 @@ module.exports = function (path, remoteApi, _remoteCall) {
     return value
   }
 
-  //add all the api methods to emitter recursively
-  ;(function recurse (obj, api, path) {
+  //add all the api methods to the emitter recursively
+  var recurse = function (obj, api, path) {
```
Comment (dominictarr) [inline api.js]: why did you change this? the self evaluating function style is better here because it's explicit that it's not called from anywhere other than inside itself.

Comment (dominictarr) [reply api.js]: oh sorry! I see it now.

File: api.js [hunk 2]
```diff
@@ -41,11 +41,25 @@ module.exports = function (path, remoteApi, _remoteCall) {
...
     })(name, api[name])
     return obj
-  })(emitter, remoteApi, path)
+  }
+
+  if (bootstrap) {
+    remoteApi['manifest'] = function () {
+      return remoteCall('async', 'manifest', [].slice.call(arguments))
+    }
+
```
Comment (dominictarr) [inline api.js]: instead of adding the method then calling it, just call `remoteCall('async', 'manifest', [cb])` in the bootstrap function

**Comment by dominictarr** [comment]
sorry, merged into 6.4.0

File: api.js (hunk 2 above)
Comment (mlegore) [reply api.js]: I don't actually use this, I don't think it's necessary, I think it was just to have the API there for if someone outside calls it.

**Comment by mlegore** [comment]
Thanks!
